Apps/Src/FaultState.c

```diff
@@ -129,26 +142,66 @@ void EnterFaultState() {
     // TODO: create an interrupt-independent CAN interface, so we can use CAN from within a fault state
     // avoid infinite recursive faults, since our CAN Driver relies on the OS to work
     // also don't call CAN if the watchdog tripped, since CAN won't be initialized
-    /*
+    
     if ((Fault_BitMap & (Fault_OS | Fault_WDOG)) == 0) {
-        CANData_t data;
-        data.b = 1;
-        CANPayload_t Message;
-        Message.data = data;
-        // Push Trip message to CAN Q
-        CANbus_BlockAndSend(TRIP, Message);
-        // Push Contactor State message to CAN Q
-        data.b = 0;
-        CANbus_BlockAndSend(CONTACTOR_STATE, Message);
+        //Deinitialize CAN registers
+        CANbus_DeInit();
+        //Reinit CAN in fault state
+        CANbus_Init(false, true);
     }
-    */
+    
 
 #ifdef DEBUGMODE
     char command[COMMAND_SIZE];
 #endif
     BSP_WDTimer_Init(); //This is in case we did not pass critical state and watchdog timer was not initialized
     BSP_WDTimer_Start(); 
+    int payload_id;
     while(1) {
+        payload_id = 0;
+        //Send Trip Message
+        CANPayload_t payload;
+        payload.idx = payload_id;
+        payload.data.w = 1;
+        CANbus_BlockAndSend_FaultState(TRIP, payload);
+        payload_id++;
+        delay_u_copy(20000);
+
+        // contactor message
+        payload.idx = payload_id;
+        payload.data.w = 2;
+        CANbus_BlockAndSend_FaultState(CONTACTOR_STATE, payload);
+        payload_id++;
+        delay_u_copy(20000);
+        
+        //Send Current Readings
+        payload.idx = payload_id;
+        payload.data.w = Amps_GetReading();
+        CANbus_BlockAndSend_FaultState(CURRENT_DATA, payload);
+        payload_id++;
+        delay_u_copy(20000);
+        
+
+        //Send Voltage Readings
+        for (int i = 0; i < NUM_BATTERY_MODULES; i++){ //send all battery module voltage data
```

Comment:
try to avoid using the ```int``` datatype -- you should be using ```uint8_t``` since all the functions you all have a ```uint8_t``` parameter and its bad to cast integer sizes down

Reply:
fixed